Samples/AudioCreation/cs/AudioCreation/Scenario7_RecordPlayback.xaml.cs

```diff
@@ -0,0 +1,589 @@
+                if (lastStatusMsec % 100 > currentStatusMsec % 100)
+                {
+                    CoreApplication.MainView.CoreWindow.Dispatcher.TryRunAsync(
+                        Windows.UI.Core.CoreDispatcherPriority.Normal,
+                        UpdateStatusWhileRecording);
+                }
+                lastStatusMsec = currentStatusMsec;
+                */
+            }
+        }
+
+        /// <summary>
+        /// Handle a frame of incoming audio from the input device.
+        /// </summary>
+        private unsafe void HandleIncomingAudio()
+        {
+            AudioFrame frame = frameOutputNode.GetFrame();
+
+            using (AudioBuffer buffer = frame.LockBuffer(AudioBufferAccessMode.Read))
+            using (IMemoryBufferReference reference = buffer.CreateReference())
+            {
+                byte* dataInBytes;
+                uint capacityInBytes;
+
+                // Get the buffer from the AudioFrame
+                ((IMemoryBufferByteAccess)reference).GetBuffer(out dataInBytes, out capacityInBytes);
+
+                if (capacityInBytes == 0)
+                {
+                    // we don't count zero-byte frames... and why do they ever happen???
+                    return;
+                }
+
+                // Must be multiple of 8 (2 channels, 4 bytes/float)
+                Debug.Assert((capacityInBytes & 0x7) == 0);
+
+                lastCapacityEncountered = capacityInBytes;
+                maxCapacityEncountered = Math.Max(maxCapacityEncountered, capacityInBytes);
+
+                uint bufferStart = 0;
+                if (maxIndexInBytes == 0)
+                {
+                    // if maxCapacityEncountered is greater than the audio graph buffer size, 
+                    // then the audio graph decided to give us a big backload of buffer content
+                    // as its first callback.  Not sure why it does this, but we don't want it,
+                    // so take only the tail of the buffer.
+                    if (maxCapacityEncountered > (graph.LatencyInSamples << 3))
+                    {
+                        bufferStart = capacityInBytes - (uint)(graph.LatencyInSamples << 3);
+                        capacityInBytes = (uint)(graph.LatencyInSamples << 3);
+                    }
+                }
+
+                // if we fill up, just spin forever.  don't try to exit recording from audio thread,
+                // we don't bother with two-way signaling.
+                if (maxIndexInBytes + capacityInBytes > byteBuffer.Length)
+                {
+                    Debug.Assert(byteBuffer.Length >= maxIndexInBytes);
+                    capacityInBytes = (uint)byteBuffer.Length - maxIndexInBytes;
+                }
+
+                fixed (byte* bufferOfBytes = byteBuffer)
+                {
+                    long* dest = ((long*)bufferOfBytes) + (maxIndexInBytes / sizeof(long));
+                    long* src = (long*)dataInBytes;
```

Comment:
Should this not be:  `long* src = (long*)dataInBytes +  bufferStart;`
for the case where you want to just copy the tail end?
currently bufferStart is assigned but not used.